chaoslib/litmus/node-drain/lib/node-drain.go

```diff
@@ -114,45 +114,65 @@ func PrepareNodeDrain(experimentsDetails *experimentTypes.ExperimentDetails, cli
 // drainNode drain the application node
 func drainNode(experimentsDetails *experimentTypes.ExperimentDetails, clients clients.ClientSets, chaosDetails *types.ChaosDetails) error {
 
-	select {
-	case <-inject:
-		// stopping the chaos execution, if abort signal received
-		os.Exit(0)
-	default:
-		log.Infof("[Inject]: Draining the %v node", experimentsDetails.TargetNode)
+	targetNodes := strings.Split(experimentsDetails.TargetNodes, ",")
+	if len(targetNodes) == 0 {
```

Comment:
Even for an empty string value in `experimentsDetails.TargetNodes`, the `len(targetNodes)` will be equal to 1. We can modify the check as follows:
```suggestion
	if experimentsDetails.TargetNodes == "" {
```